validator/client/validator.go

```diff
@@ -1159,6 +1193,11 @@ func (v *validator) ProcessEvent(event *eventClient.Event) {
 			log.WithError(err).Error("Failed to parse slot")
 		}
 		v.setHighestSlot(primitives.Slot(uintSlot))
+		if !v.disableDutiesPolling {
+			if err := v.checkDependentRoots(ctx, head, primitives.Slot(uintSlot)); err != nil {
```

Comment:
are we adding this uintSlot here just so we don't need to parse the value in head again?I feel like there's probably a better way of going about this, like maybe we update the HeadEvent with a custom Unmarshal function that will just cast the struct to the uint needed I was confused at first why we are passing in another argument here.

Reply:
also that being said ( it won't happen) but not sure what happens if line 1193 doesn't error out or return, should we make sure that we aren't calling check dependent roots if the head slot can't be parsed?

Reply:
the first message is more of a nitpick but the second part is more of a concern

Reply:
Good catch, in fact I don´t understand why we even set the highest slot in this case, it seems that this is already a bug in production